packages/client/hmi-client/src/components/data-explorer/search-bar.vue

```diff
@@ -3,127 +3,93 @@
-import { getRelatedWords, getAutocomplete } from '@/services/data';
-import Menu from 'primevue/menu';
+import AutoComplete from 'primevue/autocomplete';
 import { EventType } from '@/types/EventType';
 
-const emit = defineEmits(['query-changed', 'toggle-search-by-example']);
-
-const route = useRoute();
-const resources = useResourcesStore();
-
 const props = defineProps<{
 	suggestions: boolean;
 }>();
 
-const query = ref('');
-const autocompleteMenu = ref();
-const autocompleteMenuItems = ref([{}]);
-const inputElement = ref<HTMLInputElement | null>(null);
+const emit = defineEmits(['update-related-terms', 'toggle-search-by-example']);
+
+const route = useRoute();
+const router = useRouter();
+const resources = useResourcesStore();
 
-const isClearQueryButtonHidden = computed(() => !query.value);
+const query = ref<string>('');
+const searchBarRef = ref();
+const autocompleteMenuItems = ref<string[]>([]);
 
 function clearQuery() {
 	query.value = '';
-	emit('query-changed');
+	emit('update-related-terms');
 }
 
-const execSearch = () => {
-	emit('query-changed', query.value);
+const initiateSearch = () => {
+	emit('update-related-terms', query.value);
+	router.push({ name: RouteName.DataExplorerRoute, query: { q: query.value } });
 	EventService.create(EventType.Search, resources.activeProject?.id, query.value);
 };
 
-// const toggleSearchByExample = () => {
-// 	emit('toggle-search-by-example');
-// };
-
 function addToQuery(term: string) {
 	query.value = query.value ? query.value.trim().concat(' ').concat(term).trim() : term;
-	execSearch();
+	initiateSearch();
 	// @ts-ignore
-	inputElement.value?.$el.focus();
+	searchBarRef.value?.$el.focus();
 }
 defineExpose({ addToQuery });
 
-function replaceSearchTerm(term) {
-	query.value = term;
-	// @ts-ignore
-	inputElement.value?.$el.focus();
-}
-
-async function showAutocomplete(event) {
-	if (query.value.length >= 3) {
-		const promise = getAutocomplete(query.value);
-		promise.then((response) => {
-			autocompleteMenuItems.value = response.map((item) => ({
-				label: item,
-				icon: 'pi pi-search',
-				command: () => {
-					replaceSearchTerm(item);
-				}
-			}));
-			// @ts-ignore
-			inputElement.value?.$el.focus();
-		});
-		autocompleteMenu.value.show(event);
-	}
-}
-
-async function showSuggestions(event) {
-	if (props.suggestions) {
-		const promise = getRelatedWords(query.value, resources.xddDataset);
-		promise.then((response) => {
-			autocompleteMenuItems.value = response.map((item) => ({
-				label: item,
-				icon: 'pi pi-search',
-				command: () => {
-					addToQuery(item);
-				}
-			}));
-			// @ts-ignore
-			inputElement.value?.$el.focus();
-		});
-		autocompleteMenu.value.show(event);
-	}
-}
-
-const handleSearchEvent = (event) => {
+const fillAutoComplete = async (event) => {
 	const keyboardEvent = event as KeyboardEvent;
-	if (keyboardEvent.code === 'Space') {
-		showSuggestions(event);
+	let promise: Promise<string[]>;
```

Comment:
Nitpick: good place to use a ternary. 

`const promise: Promise<string[]> = (keyboardEvent.code === 'Space' && props.suggestions) ?
getRelatedTerms(query.value, resources.xddDataset) : 
getAutocomplete(query.value)`